From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nds-ports <nds-ports@localhost>
Date: Tue, 6 Oct 2026 07:09:57 +0200
Subject: [PATCH] Use single precision sine and cosine

The ARM9 has no floating point unit, AngleVectors, which runs for every
entity, uses the float versions of sin and cos, which are a lot cheaper
in software than the double ones.
---
 WinQuake/mathlib.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/WinQuake/mathlib.c b/WinQuake/mathlib.c
index 9f5d644..c70385b 100644
--- a/WinQuake/mathlib.c
+++ b/WinQuake/mathlib.c
@@ -289,18 +289,22 @@ if (sides == 0)
 #endif
 
 
+// without an FPU the float versions are a lot cheaper than the double ones
+#define	sin(x)	sinf(x)
+#define	cos(x)	cosf(x)
+
 void AngleVectors (vec3_t angles, vec3_t forward, vec3_t right, vec3_t up)
 {
 	float		angle;
 	float		sr, sp, sy, cr, cp, cy;
 	
-	angle = angles[YAW] * (M_PI*2 / 360);
+	angle = angles[YAW] * (float)(M_PI*2 / 360);
 	sy = sin(angle);
 	cy = cos(angle);
-	angle = angles[PITCH] * (M_PI*2 / 360);
+	angle = angles[PITCH] * (float)(M_PI*2 / 360);
 	sp = sin(angle);
 	cp = cos(angle);
-	angle = angles[ROLL] * (M_PI*2 / 360);
+	angle = angles[ROLL] * (float)(M_PI*2 / 360);
 	sr = sin(angle);
 	cr = cos(angle);
 
